prime-angular-frontend/src/app/modules/enrolment/pages/pharmanet-enrolment-summary/pharmanet-enrolment-summary.component.html

```diff
@@ -17,37 +17,10 @@
 
         <form [formGroup]="form">
 
-          <ng-container *ngFor="let careSetting of careSettings">
-
-            <ng-container [ngSwitch]="careSetting.careSettingCode">
-
-              <ng-container *ngSwitchCase="CareSettingEnum.PRIVATE_COMMUNITY_HEALTH_PRACTICE"
-                            [ngTemplateOutlet]="settingEmail"
-                            [ngTemplateOutletContext]="{
-                              setting: 'Private Community Health Practice',
-                              settingCode: careSetting.careSettingCode,
-                              formControl: communityHealthEmails
-                            }">
-              </ng-container>
-
-              <ng-container *ngSwitchCase="CareSettingEnum.COMMUNITY_PHARMACIST"
-                            [ngTemplateOutlet]="settingEmail"
-                            [ngTemplateOutletContext]="{
-                              setting: 'Community Pharmacy',
-                              settingCode: careSetting.careSettingCode,
-                              formControl: pharmacistEmails
-                            }">
-              </ng-container>
-
-              <ng-container *ngSwitchCase="CareSettingEnum.HEALTH_AUTHORITY"
-                            [ngTemplateOutlet]="settingEmail"
-                            [ngTemplateOutletContext]="{
-                              setting: 'Health Authority',
-                              settingCode: careSetting.careSettingCode,
-                              formControl: healthAuthorityEmails
-                            }">
-              </ng-container>
+          <ng-container *ngFor="let careSettingConfig of careSettingConfigs">
```

Comment:
There is absolutely nothing wrong with using directives like you were, but keeping the logic in the controller and using configuration objects reduces the complexity of the template.  This is super readable with the logic in the controller.